domain/servicefactory/model.go

```diff
@@ -66,6 +70,17 @@ func NewModelFactory(
 	}
 }
 
+// AgentProvisioner returns the agent provisioner service.
+func (s *ModelFactory) AgentProvisioner() *agentprovisionerservice.Service {
+	return agentprovisionerservice.NewService(
+		s.modelUUID,
+		agentprovisionerstate.NewState(
+			changestream.NewTxnRunnerFactory(s.modelDB),
+		),
+		providertracker.ProviderRunner[agentprovisionerservice.Provider](s.providerFactory, s.modelUUID.String()),
```

Comment:
Because it allows you to see other models. It's required in some code paths. We could special case those cases?

Reply:
I agree with @wallyworld here. I think the `ProviderRunner` shouldn't require it to be passed in the common case. It can lead to issues around divergent `modelUUID` for models as a programmatic error.

Reply:
The provider runner has already been designed this way, changing it is out of scope for this PR.

Happy to change it in a future PR and update this usage.